### PR TITLE
Add BaseProps to all components

### PR DESCRIPTION
**Pull request by nktssh**

### CODE REVIEW

File: src/components/TextField/TextField.tsx [hunk 1]
```diff
@@ -60,12 +62,16 @@ export const TextField: React.FunctionComponent<TextFieldProps> = (
...
-      })}
+      {...style(
+        'root',
+        {
+          error,
+          theme,
+          success,
+          disabled,
+        },
+        { className },
```
Comment (jonathanadler) [inline src/components/TextField/TextField.tsx]: instead of passing just the className, lets pass `...rest`

File: src/types.ts [hunk 2]
```diff
@@ -0,0 +1,3 @@
+export interface TPAComponentProps {
+  className?: string;
+}
```
Comment (jonathanadler) [inline src/types.ts]: Let's also add
```
'data-hook'?: string;
```

File: src/components/Input/Input.tsx [hunk 3]
```diff
@@ -5,8 +5,9 @@ import {
 } from 'wix-ui-core/input';
 import style from './Input.st.css';
 import { ErrorMessageWrapper, ErrorProps } from '../ErrorMessageWrapper';
+import { TPAComponentProps } from '../../types';
 
-export interface TPAInputProps {
+export interface TPAInputProps extends TPAComponentProps {
```
Comment (jonathanadler) [inline src/components/Input/Input.tsx]: This component is theoretically deprecated, so let's not change it.

Comment (jonathanadler) [reply src/components/Input/Input.tsx]: I still prefer we don't change this

**Comment by nktssh** [comment]
Rolled `Input` back

**Pull request merged**